src/client/automation/playback/scroll.js

```diff
@@ -267,24 +265,21 @@ export default class ScrollAutomation {
             });
     }
 
-    _isElementObscured (element) {
-        let el           = element;
-        let fixedElement = null;
-
-        while (el && !fixedElement) {
-            if (styleUtils.isFixedElement(el))
-                fixedElement = el;
-
-            el = el.parentNode;
-        }
+    static _getFixedAncestorOrSelf (element) {
+        const ancestors = domUtils.getAncestorsAndSelf(element).reverse();
 
-        return element && fixedElement && !fixedElement.contains(this.element);
+        return arrayUtils.find(ancestors, el => styleUtils.isFixedElement(el));
```

Comment:
Previous variant was faster. 
I noticed we always use `getAncestorsAndSelf` and then find some element by predicate (visible, hidden, fixed), what if we add parameter: `getAncestorsAndSelf(node, predicate)`

Reply:
Yes, it was faster but I'd prefer to use already written method to prevent duplicates. Add predicate param is nice idea.